### PR TITLE
More accurate scanner identification

### PR DESCRIPTION
**Pull request by gpotter2**
Detection of some scanners:
- detects nmap when it uses a default list of 1000 ports (needs a double check against other scanners but should be fine)
- add an nmap UDP probe
- detect some TCP probe packets from Nmap (requires https://github.com/zeek/zeek/pull/1769 so I'm not sure if this PR is mergeable until a new version is released :/)
- a few improvements to how scanners are displayed (now also displays which probe were found to be able to identify false positives, if any)

### CODE REVIEW

File: ivre/active/data.py [hunk 1]
```diff
@@ -28,9 +28,9 @@
 from textwrap import wrap
 from typing import Any, Callable, Dict, List, Set, Union, cast
 
-
```
Comment (p-l-) [inline ivre/active/data.py:31]: Please try to avoid unneeded changes if possible

**Review by p-l- — COMMENTED** [review]
Thanks for this PR, that's great!

Would you mind splitting it into small PRs? (maybe, one per item from your list?

File: ivre/data/scanners.py [hunk 2]
```diff
@@ -57,10 +57,15 @@
         "Censys",
         "DNS",
     ),  # DNS query IN PTR 213.1.168.192.in-addr.arpa.  # noqa: E501
+    # https://nmap.org/book/osdetect-methods.html
+    "CCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCC": (
```
Comment (p-l-) [inline ivre/data/scanners.py:61]: Should be bytes I guess?

File: ivre/data/scanners.py [hunk 3]
```diff
@@ -70,3 +75,23 @@
...
+    # https://docs.zeek.org/en/v4.0.0/scripts/base/init-bare.zeek.html#type-SYN_packet
+    # https://nmap.org/book/osdetect-methods.html
+
+    # Format: (DF, win_size, win_scale, MSS, SACK_OK, TSval, TSecr): ("Scanner name", "Probe name")
+    (0, 4, 5, 640, 0, 0xFFFFFFFF, 0): ("Nmap", "Fingerprint-P3"),
+    (1, 31337, 10, 265, 1, 0xFFFFFFFF, 0): ("Nmap", "Fingerprint-T5")
+}
+
+# ports that scanners scan by default
+DEFAULT_SCANNED_PORTS = {
```
Comment (p-l-) [inline ivre/data/scanners.py:90]: Neat idea! You may consider adding top 100 (-F option in Nmap)?

**Comment by gpotter2** [comment]
Split in:
- https://github.com/ivre/ivre/pull/1178
- https://github.com/ivre/ivre/pull/1179
- https://github.com/ivre/ivre/pull/1180

**Pull request closed without merging**